src/components/profile/editProfile.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Button, Form, FormGroup, Label, Input, FormText } from "reactstrap";
+import PropTypes from "prop-types";
+import { editProfile } from "../../actions/profileActions/profileActions";
+import decode from "jwt-decode";
+import NavBar from "../../components/navigation/navBar";
+
+class EditProfile extends Component {
+  static propTypes = {
+    dispatch: PropTypes.func,
+    history: PropTypes.object,
+    profile: PropTypes.object
+
+  };
+  constructor(props) {
+    super(props);
+    const token = localStorage.getItem("token");
+    let details = decode(token);
+    const user = details.username;
+    this.state = {
+      profile: {
+        username: user,
+        first_name: "",
+        last_name: "",
+        bio: ""
+      },
+      refer: false
+    };
+  }
+
+  handleInputChange = event => {
+    const target = event.target;
+    const value = event.target.value;
+    const name = target.name;
+
+    this.setState(prevState => {
+      return {
+        profile: {
+          ...prevState.profile,
+          [name]: value
+        }
+      };
+    });
+  };
+
+  componentWillReceiveProps(nextProps) {
+    let username = nextProps.profile.username;
+    if (username) {
+      this.props.history.push(`/profile/${username}`);
+    }
+  }
+  handleEditProfile = event => {
+    event.preventDefault();
+    let data = { profile: this.state.profile };
+    this.props.dispatch(editProfile(this.state.profile.username, data));
+  };
+  render() {
+    return (
+      <>
```

Comment:
Create a views component in the views folder that will hold the jsx and simply call the component here, the same way you did for the view profile component

Reply:
I have changed it and created a view componet